### PR TITLE
feat(sgcspevaluatorcli): add csp helper type

### PR DESCRIPTION
**Pull request by ngalaiko**
ContentSecurityPolicy type provides a set of helper functions to define content security policy in a structured way.

### CODE REVIEW

File: tools/sgcspevaluatorcli/csp.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	cloned := map[string][]string{}
+	for key, values := range csp {
+		cloned[key] = make([]string, len(values))
+		copy(cloned[key], values)
+	}
+	return cloned
+}
+
+// With returns a new content security policy with an additional directive and sources.
+func (csp ContentSecurityPolicy) With(directive, source string, sources ...string) ContentSecurityPolicy {
```
Comment (odsod) [inline tools/sgcspevaluatorcli/csp.go:35]: how come we separate out the first vararg to a separate arg here?

Comment (ngalaiko) [reply tools/sgcspevaluatorcli/csp.go:35]: that's to force at least one source on the compilation level.

i.e.
```
csp.With("directive")
```

won't compile

Comment (odsod) [reply tools/sgcspevaluatorcli/csp.go:35]: ah! I see, nice way of avoiding errors/panics!

File: tools/sgcspevaluatorcli/csp.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	for key, values := range csp {
+		cloned[key] = make([]string, len(values))
+		copy(cloned[key], values)
+	}
+	return cloned
+}
+
+// With returns a new content security policy with an additional directive and sources.
+//
+// Note that the _new_ policy is returned here, so changing the returned value from this function will
```
Comment (ngalaiko) [inline tools/sgcspevaluatorcli/csp.go:36]: @radhus i've added this note. does this make it clear why cloning is there?

Comment (radhus) [reply tools/sgcspevaluatorcli/csp.go:36]: Yes 👍

**Review by radhus — APPROVED** [review]

**Review by odsod — APPROVED** [review]
Good stuff!

**Pull request merged**